notifications-sw.js

```diff
@@ -0,0 +1,44 @@
+/*!
+ * Matomo - free/libre analytics platform
+ *
+ * UsersManager screenshot tests.
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+var token;
+
+self.addEventListener('install', async function() {
+    self.token = new URL(location).searchParams.get('token');
+});
+
+self.addEventListener('activate', async function() {
+    // Fetch any notifications that have already been generated
+    fetchNotifications();
+
+    // Check periodically for new notifications
+    this.timerId = setInterval(function() {
+        fetchNotifications();
+    }, 30 * 1000);  // Timeout is in ms
+});
+
+var fetchNotifications = function() {
+    fetch('index.php?module=MobileMessaging&action=getBrowserNotifications&token=' + self.token, 
+        {credentials: "omit"}
+    ).then(function(response) {
+        if (response.status === 200) {
```

Comment:
I reckon a 301 or 302 might be fine as well? We've seen users sometimes redirect these requests. Although in this case shouldn't happen as much. 

FYI: I know some users have basic auth etc set up. We may need to mention in an FAQ that it won't work in this case.

Because we will get reports from users that for some reason it doesn't work, can we log errors to the console? Eg status code, message, ... ?